src/graph/visitor/DeduceAliasTypeVisitor.cpp

```diff
@@ -0,0 +1,91 @@
+/* Copyright (c) 2022 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License.
+ */
+
+#include "graph/visitor/DeduceAliasTypeVisitor.h"
+
+#include <sstream>
+#include <unordered_map>
+
+#include "common/datatypes/DataSet.h"
+#include "common/datatypes/Edge.h"
+#include "common/datatypes/List.h"
+#include "common/datatypes/Map.h"
+#include "common/datatypes/Path.h"
+#include "common/datatypes/Set.h"
+#include "common/function/FunctionManager.h"
+#include "graph/context/QueryContext.h"
+#include "graph/context/QueryExpressionContext.h"
+#include "graph/context/ValidateContext.h"
+#include "graph/visitor/EvaluableExprVisitor.h"
+
+namespace nebula {
+namespace graph {
+
+DeduceAliasTypeVisitor::DeduceAliasTypeVisitor(QueryContext *qctx,
+                                               ValidateContext *vctx,
+                                               GraphSpaceID space,
+                                               AliasType inputType)
+    : qctx_(qctx), vctx_(vctx), space_(space), inputType_(inputType) {
+  UNUSED(qctx_);
+  UNUSED(vctx_);
+  UNUSED(space_);
+}
+
+void DeduceAliasTypeVisitor::visit(VertexExpression *expr) {
+  UNUSED(expr);
+  outputType_ = AliasType::kNode;
+}
+
+void DeduceAliasTypeVisitor::visit(EdgeExpression *expr) {
+  UNUSED(expr);
+  outputType_ = AliasType::kEdge;
+}
+
+void DeduceAliasTypeVisitor::visit(PathBuildExpression *expr) {
+  UNUSED(expr);
+  outputType_ = AliasType::kPath;
+}
+
+void DeduceAliasTypeVisitor::visit(FunctionCallExpression *expr) {
+  auto funName = expr->name();
+  if (funName == "nodes") {
+    outputType_ = AliasType::kNodeList;
+  } else if (funName == "relationships") {
+    outputType_ = AliasType::kEdgeList;
+  } else if (funName == "reversepath") {
+    outputType_ = AliasType::kPath;
+  } else if (funName == "startnode" || funName == "endnode") {
+    outputType_ = AliasType::kNode;
+  }
```

Comment:
There may be case-sensitive issues with function names.

Reply:
fixed